src/projections.jl

```diff
@@ -152,6 +158,117 @@ function vec_symm(X)
     return X[LinearAlgebra.triu(trues(size(X)))]
 end
 
+"""
+    projection_on_set(::DefaultDistance, v::AbstractVector{T}, ::MOI.ExponentialCone) where {T}
+
+projection of vector `v` on closure of the exponential cone
+i.e. `cl(Kexp) = {(x,y,z) | y e^(x/y) <= z, y>0 } U {(x,y,z)| x <= 0, y = 0, z >= 0}`.
+
+References:
+* [Proximal Algorithms, 6.3.4](https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf)
+by Neal Parikh and Stephen Boyd.
+* [Projection, presolve in MOSEK: exponential, and power cones]
+(https://docs.mosek.com/slides/2018/ismp2018/ismp-friberg.pdf) by Henrik Friberg
+"""
+function projection_on_set(::DefaultDistance, v::AbstractVector{T}, ::MOI.ExponentialCone) where {T}
+    # https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf, Section 6.3
+    length(v) != 3 && throw(DimensionMismatch("Mismatch between value and set"))
```

Comment:
```suggestion
    _check_dimension(v, s)
```
this should be used at the beginning, specifically meant for that :)

Reply:
Unresolving so I remember to add `s` to the function definition